rotkehlchen/globaldb/handler.py

```diff
@@ -2046,34 +2092,27 @@ def add_location_asset_mappings(
 
         May Raise (if skip_errors is False):
         - InputError if any of the pairs of location and exchange_symbol already exist"""
-        with GlobalDBHandler().conn.write_ctx() as cursor:
-            for entry in entries:
-                try:
-                    if entry.location is None and cursor.execute(
-                        'SELECT COUNT(*) FROM location_asset_mappings WHERE location IS NULL AND exchange_symbol=? AND local_id=?',  # noqa: E501
-                        (entry.location_symbol, entry.asset.serialize()),
-                    ).fetchone()[0] > 0:
-                        raise sqlite3.IntegrityError('Entry already exists in the DB')
-
+        def location_null_precheck(cursor: 'DBCursor', entry: LocationAssetMappingUpdateEntry) -> None:  # noqa: E501
```

Comment:
define that outside the function. Inner functions are redefined every time you call the parent.